### ISSUE TITLE
Support anchor in pixels 

### ISSUE THREAD

**Issue by mattjennings** [excaliburjs/Excalibur#2923]
### Context

Sometimes it's helpful to be able to define an Actor's anchor in pixels instead of %. e.g. if your sprite isn't perfectly centered on the frame and the anchor point would be 8px down from the middle, trying to find the % for that exact offset is  tricky.

### Proposal

Two proposals:

## an Anchor class that defines the unit

My initial idea when writing this issue was an `Anchor` class that could be used as the `anchor` argument.

```ts
new ex.Actor({
  anchor: new ex.Anchor(x,y, 'px' | 'relative')
})
``` 

However it's ambiguous what this offset would be relative to. I suppose top-left is what it would be, but there are times you might want it to be offset from the center (but that just means some extra math based on width/height, so maybe it's not a big deal).

## an anchorOffset argument

If there were an additional argument `anchorOffset` that would offset in px from the `anchor` value, that would give you the flexibility to easily shift things around as needed. I just don't like having 2 different properties for this. It could cause confusion.

**Comment by eonarheim** [comment]
@mattjennings I agree! I think a pixel offset is useful in this case. Let me play around with the first one, a `ex.Anchor` type that extends vector would be a relatively easy drop in.

We do have a way to do this (but not super visible on actor). On the graphics component we have a `actor.graphics.offset` property that will shift graphics by pixels.

```typescript
const actor = new ex.Actor({...});
actor.graphics.anchor = ex.vec(0, 0); // zero out the anchor
actor.graphics.offset = ex.vec(-8, -8); // offset 
```

Internally this is the drawing code in GraphicsSystem

```typescript
    let anchor = graphicsComponent.anchor;
    let offset = graphicsComponent.offset;

    // handle specific overrides
    if (options?.anchor) {
      anchor = options.anchor;
    }
    if (options?.offset) {
      offset = options.offset;
    }
    // See https://github.com/excaliburjs/Excalibur/pull/619 for discussion on this formula
    const offsetX = -graphic.width * anchor.x + offset.x;
    const offsetY = -graphic.height * anchor.y + offset.y;

    const oldFlipHorizontal = graphic.flipHorizontal;
    const oldFlipVertical = graphic.flipVertical;
    if (flipHorizontal || flipVertical) {
      // flip any currently flipped graphics
      graphic.flipHorizontal = flipHorizontal ? !oldFlipHorizontal : oldFlipHorizontal;
      graphic.flipVertical = flipVertical ? !oldFlipVertical : oldFlipVertical;
    }

    graphic?.draw(
      this._graphicsContext,
      offsetX,
      offsetY);
```

**Comment by eonarheim** [comment]
We could also expose `offset` in the actor constructor?

**Comment by mattjennings** [comment]
I actually did try using the offset on the graphics initially (and I do think it's worth exposing that on the actor constructor), but it wasn't quite the same thing as adjusting the anchor. For example, when I applied a scale it still scaled according to the anchor point. I suppose the scale effect could account for this, but I think this more a symptom of the issue.

**Comment by eonarheim** [comment]
Good points, also it should totally account for the scale that's definitely a bug
